contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -4648,8 +4729,30 @@ get_bbf_pivot_tuplestore(RawStmt 	*sql,
 		/* internal error */
 		elog(ERROR, "get_bbf_pivot_tuplestore: SPI_connect returned %d", ret);
 
-	/* Now retrieve the bbf_pivot source rows */
-	ret = SPI_execute_raw_parsetree(sql, sourcetext, true, 0);
+	/* 
+	 * Now retrieve the bbf_pivot source rows    
+	 *
+	 * If view_uuid is defined, then we will use the provided uuid to form the source_sql and 
+	 * get result. source_sql name format is pvt_sv_[uuid]
+	 * 
+	 * if view_uuid is not defined, then we will use the rawparsetree of source_sql to get the
+	 * result
+	 */
+	if (!view_uuid || strlen(view_uuid) != 0)
```

Comment:
shouldn't this be `if (view_uuid && strlen(view_uuid) != 0)`?

Reply:
my dumb mistake again 😥